### PR TITLE
Add gamestate to player

### PR DESCRIPTION
**Pull request by yummyblabla**
Add gamestate (wind and flowernumber) to Player Hand for hand/point validation

### CODE REVIEW

File: src/modules/mahjong/Hand/test/PlayerHand.test.ts [hunk 1]
```diff
@@ -93,3 +94,38 @@ test('PlayerHand - draw()', () => {
   expect(fullHand.getTiles()).toHaveLength(13);
   expect(fullHand.getTiles()).toStrictEqual(tiles);
 });
+
+test('PlayerHand - getFlowerNumber()', () => {
+  expect(emptyHand.getFlowerNumber()).toBe(1);
+  expect(fullHand.getFlowerNumber()).toBe(1);
+});
+
+test('PlayerHand - setFlowerNumber()', () => {
```
Comment (vwong125) [inline src/modules/mahjong/Hand/test/PlayerHand.test.ts:103]: kind of triggered because was taught to only have 1 expect for each test case

Comment (jzc1828) [reply src/modules/mahjong/Hand/test/PlayerHand.test.ts:103]: Really? During your coop?

Comment (yummyblabla) [reply src/modules/mahjong/Hand/test/PlayerHand.test.ts:103]: Interesting :o @jzc1828 what did you do?

I never did much testing so this is what I've been doing without any guidance

Comment (jzc1828) [reply src/modules/mahjong/Hand/test/PlayerHand.test.ts:103]: I usually do have a few `expect` to check the returned value of the function being tested and if the function calls some other function(s) within it, I will have `expect` for the function calls:
```ts
expect(funcSpy).toHaveBeenCalledTimes(numofTimes);
expect(funcSpy).toHaveBeenCalledWith(arguments);
```

Comment (vwong125) [reply src/modules/mahjong/Hand/test/PlayerHand.test.ts:103]: Yea was taught to have 1 expect per test case to ensure modularity. If you have multiple expects in a test case, you wouldn't know which expect statement fails during the test.

**Pull request merged**